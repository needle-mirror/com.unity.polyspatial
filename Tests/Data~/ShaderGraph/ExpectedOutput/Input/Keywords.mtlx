<?xml version ="1.0"?>
<materialx version = "1.38" colorspace = "lin_rec709">
    <ifequal name = "_1" type = "float">
        <input name = "value1" type = "boolean" nodename = "_ENUM_A" />
        <input name = "value2" type = "boolean" value = "true" />
        <input name = "in1" type = "float" nodename = "_2" />
        <input name = "in2" type = "float" nodename = "_3" />
    </ifequal>
    <constant name = "_ENUM_A" type = "boolean">
        <input name = "value" type = "boolean" value = "false" />
    </constant>
    <dot name = "_2" type = "float">
        <input name = "in" type = "float" value = "0" />
    </dot>
    <ifequal name = "_3" type = "float">
        <input name = "value1" type = "boolean" nodename = "_ENUM_B" />
        <input name = "value2" type = "boolean" value = "true" />
        <input name = "in1" type = "float" nodename = "_4" />
        <input name = "in2" type = "float" nodename = "_5" />
    </ifequal>
    <constant name = "_ENUM_B" type = "boolean">
        <input name = "value" type = "boolean" value = "false" />
    </constant>
    <dot name = "_4" type = "float">
        <input name = "in" type = "float" value = "1" />
    </dot>
    <dot name = "_5" type = "float">
        <input name = "in" type = "float" value = "0" />
    </dot>
    <constant name = "_BOOLEAN" type = "boolean">
        <input name = "value" type = "boolean" value = "false" />
    </constant>
    <ifequal name = "_6" type = "float">
        <input name = "in1" type = "float" value = "1" />
        <input name = "in2" type = "float" value = "0" />
        <input name = "value1" type = "boolean" nodename = "_BOOLEAN" />
        <input name = "value2" type = "boolean" value = "true" />
    </ifequal>
    <subtract name = "_7" type = "float">
        <input name = "in1" type = "float" nodename = "_1" />
        <input name = "in2" type = "float" nodename = "_6" />
    </subtract>
    <dot name = "_8" type = "float">
        <input name = "in" type = "float" nodename = "_9" />
    </dot>
    <ifequal name = "_9" type = "float">
        <input name = "value1" type = "float" nodename = "_10" />
        <input name = "value2" type = "float" value = "0" />
        <input name = "in1" type = "float" value = "0" />
        <input name = "in2" type = "float" value = "1" />
    </ifequal>
    <add name = "_10" type = "float">
        <input name = "in1" type = "float" nodename = "_11" />
        <input name = "in2" type = "float" nodename = "_13" />
    </add>
    <absval name = "_11" type = "float">
        <input name = "in" type = "float" nodename = "_12" />
    </absval>
    <dot name = "_12" type = "float">
        <input name = "in" type = "float" nodename = "_1" />
    </dot>
    <absval name = "_13" type = "float">
        <input name = "in" type = "float" nodename = "_14" />
    </absval>
    <dot name = "_14" type = "float">
        <input name = "in" type = "float" nodename = "_6" />
    </dot>
    <ifgreater name = "_15" type = "float">
        <input name = "in1" type = "float" nodename = "_7" />
        <input name = "in2" type = "float" value = "1" />
        <input name = "value1" type = "float" nodename = "_16" />
    </ifgreater>
    <absval name = "_16" type = "float">
        <input name = "in" type = "float" nodename = "_8" />
    </absval>
    <dot name = "_17" type = "vector4">
        <input name = "in" type = "vector4" nodename = "_18" />
    </dot>
    <combine4 name = "_18" type = "vector4">
        <input name = "in1" type = "float" nodename = "_19" />
        <input name = "in2" type = "float" nodename = "_20" />
        <input name = "in3" type = "float" nodename = "_21" />
        <input name = "in4" type = "float" nodename = "_22" />
    </combine4>
    <dot name = "_19" type = "float">
        <input name = "in" type = "float" nodename = "_28" />
    </dot>
    <dot name = "_20" type = "float">
        <input name = "in" type = "float" nodename = "_27" />
    </dot>
    <dot name = "_21" type = "float">
        <input name = "in" type = "float" value = "0" />
    </dot>
    <dot name = "_22" type = "float">
        <input name = "in" type = "float" value = "0" />
    </dot>
    <dot name = "_23" type = "vector3">
        <input name = "in" type = "vector3" nodename = "_24" />
    </dot>
    <combine3 name = "_24" type = "vector3">
        <input name = "in1" type = "float" nodename = "_19" />
        <input name = "in2" type = "float" nodename = "_20" />
        <input name = "in3" type = "float" nodename = "_21" />
    </combine3>
    <dot name = "_25" type = "vector2">
        <input name = "in" type = "vector2" nodename = "_26" />
    </dot>
    <combine2 name = "_26" type = "vector2">
        <input name = "in1" type = "float" nodename = "_19" />
        <input name = "in2" type = "float" nodename = "_20" />
    </combine2>
    <subtract name = "_27" type = "float">
        <input name = "in2" type = "float" nodename = "_15" />
        <input name = "in1" type = "float" value = "1" />
    </subtract>
    <ifgreater name = "_28" type = "float">
        <input name = "in1" type = "float" nodename = "_7" />
        <input name = "in2" type = "float" value = "1" />
        <input name = "value1" type = "float" nodename = "_29" />
    </ifgreater>
    <absval name = "_29" type = "float">
        <input name = "in" type = "float" nodename = "_8" />
    </absval>
    <GeometryModification name = "_30" type = "vertexshader">
    </GeometryModification>
    <realitykit_unlit name = "_31" type = "surfaceshader">
        <input name = "applyPostProcessToneMap" type = "boolean" value = "false" />
        <input name = "color" type = "color3" nodename = "_32" />
    </realitykit_unlit>
    <surfacematerial name = "USD_Keywords" type = "material">
        <input name = "vertexshader" type = "vertexshader" nodename = "_30" />
        <input name = "surfaceshader" type = "surfaceshader" nodename = "_31" />
    </surfacematerial>
    <swizzle name = "_32" type = "color3">
        <input name = "in" type = "vector3" nodename = "_23" />
        <input name = "channels" type = "string" value = "xyz" />
    </swizzle>
</materialx>
