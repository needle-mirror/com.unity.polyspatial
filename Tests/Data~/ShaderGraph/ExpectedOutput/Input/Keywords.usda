#usda 1.0
(
    metersPerUnit = 1
    upAxis = "Y"
)

def "MaterialX"
{
    def "Materials"
    {
        def Material "USD_Keywords"
        {
            bool inputs:_ENUM_A = 0
            bool inputs:_ENUM_B = 0
            bool inputs:_BOOLEAN = 0

            token outputs:mtlx:surface.connect = <realitykit_unlit.outputs:out>
            token outputs:realitykit:vertex.connect = <GeometryModifier.outputs:out>

            def Shader "realitykit_unlit"
            {
                uniform token info:id = "ND_realitykit_unlit_surfaceshader"
                bool inputs:applyPostProcessToneMap = 0
                color3f inputs:color.connect = <../_32.outputs:out>
                token outputs:out
            }

            def Shader "GeometryModifier"
            {
                uniform token info:id = "ND_realitykit_geometrymodifier_vertexshader"
                token outputs:out
            }

            def Shader "_32"
            {
                uniform token info:id = "ND_swizzle_vector3_color3"
                float3 inputs:in.connect = <../_23.outputs:out>
                string inputs:channels = "xyz"
                color3f outputs:out
            }

            def Shader "_23"
            {
                uniform token info:id = "ND_dot_vector3"
                float3 inputs:in.connect = <../_24.outputs:out>
                float3 outputs:out
            }

            def Shader "_24"
            {
                uniform token info:id = "ND_combine3_vector3"
                float inputs:in1.connect = <../_19.outputs:out>
                float inputs:in2.connect = <../_20.outputs:out>
                float inputs:in3.connect = <../_21.outputs:out>
                float3 outputs:out
            }

            def Shader "_19"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_28.outputs:out>
                float outputs:out
            }

            def Shader "_28"
            {
                uniform token info:id = "ND_ifgreater_float"
                float inputs:in1.connect = <../_7.outputs:out>
                float inputs:in2 = 1
                float inputs:value1.connect = <../_29.outputs:out>
                float outputs:out
            }

            def Shader "_7"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in1.connect = <../_1.outputs:out>
                float inputs:in2.connect = <../_6.outputs:out>
                float outputs:out
            }

            def Shader "_1"
            {
                uniform token info:id = "ND_ifequal_floatB"
                bool inputs:value1.connect = <../.inputs:_ENUM_A>
                bool inputs:value2 = 1
                float inputs:in1.connect = <../_2.outputs:out>
                float inputs:in2.connect = <../_3.outputs:out>
                float outputs:out
            }

            def Shader "_2"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in = 0
                float outputs:out
            }

            def Shader "_20"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_27.outputs:out>
                float outputs:out
            }

            def Shader "_27"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in2.connect = <../_15.outputs:out>
                float inputs:in1 = 1
                float outputs:out
            }

            def Shader "_15"
            {
                uniform token info:id = "ND_ifgreater_float"
                float inputs:in1.connect = <../_7.outputs:out>
                float inputs:in2 = 1
                float inputs:value1.connect = <../_16.outputs:out>
                float outputs:out
            }

            def Shader "_16"
            {
                uniform token info:id = "ND_absval_float"
                float inputs:in.connect = <../_8.outputs:out>
                float outputs:out
            }

            def Shader "_8"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_9.outputs:out>
                float outputs:out
            }

            def Shader "_9"
            {
                uniform token info:id = "ND_ifequal_float"
                float inputs:value1.connect = <../_10.outputs:out>
                float inputs:value2 = 0
                float inputs:in1 = 0
                float inputs:in2 = 1
                float outputs:out
            }

            def Shader "_10"
            {
                uniform token info:id = "ND_add_float"
                float inputs:in1.connect = <../_11.outputs:out>
                float inputs:in2.connect = <../_13.outputs:out>
                float outputs:out
            }

            def Shader "_13"
            {
                uniform token info:id = "ND_absval_float"
                float inputs:in.connect = <../_14.outputs:out>
                float outputs:out
            }

            def Shader "_14"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_6.outputs:out>
                float outputs:out
            }

            def Shader "_11"
            {
                uniform token info:id = "ND_absval_float"
                float inputs:in.connect = <../_12.outputs:out>
                float outputs:out
            }

            def Shader "_12"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_1.outputs:out>
                float outputs:out
            }

            def Shader "_21"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in = 0
                float outputs:out
            }

            def Shader "_29"
            {
                uniform token info:id = "ND_absval_float"
                float inputs:in.connect = <../_8.outputs:out>
                float outputs:out
            }

            def Shader "_6"
            {
                uniform token info:id = "ND_ifequal_floatB"
                float inputs:in1 = 1
                float inputs:in2 = 0
                bool inputs:value1.connect = <../.inputs:_BOOLEAN>
                bool inputs:value2 = 1
                float outputs:out
            }

            def Shader "_3"
            {
                uniform token info:id = "ND_ifequal_floatB"
                bool inputs:value1.connect = <../.inputs:_ENUM_B>
                bool inputs:value2 = 1
                float inputs:in1.connect = <../_4.outputs:out>
                float inputs:in2.connect = <../_5.outputs:out>
                float outputs:out
            }

            def Shader "_4"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in = 1
                float outputs:out
            }

            def Shader "_5"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in = 0
                float outputs:out
            }
        }
    }
}
