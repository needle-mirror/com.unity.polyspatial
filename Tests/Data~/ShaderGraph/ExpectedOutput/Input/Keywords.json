{"properties":[{"name":"_ENUM_A","type":0},{"name":"_ENUM_B","type":0},{"name":"_BOOLEAN","type":0}]}